src/WtsApi32.Desktop/WtsApi32.cs

```diff
@@ -0,0 +1,112 @@
+        /// will have a session identifier of zero (0). You can use the WTSEnumerateSessions function to retrieve the identifiers of
+        /// all sessions on a specified RD Session Host server.
+        /// To be able to query information for another user's session, you need to have the Query Information permission.
+        /// For more information, see Remote Desktop Services Permissions. To modify permissions on a session, use the
+        /// Remote Desktop Services Configuration administrative tool.</param>
+        /// <param name="phToken">If the function succeeds, receives a pointer to the token handle for the logged-on user. Note
+        /// that you must call the CloseHandle function to close this handle.</param>
+        /// <returns>f the function succeeds, the return value is a nonzero value, and the phToken parameter points to the primary
+        /// token of the user.</returns>
+        [DllImport(nameof(WtsApi32), SetLastError = true)]
+        public static extern bool WTSQueryUserToken(int SessionId, out SafeObjectHandle phToken);
+
+        /// <summary>
+        /// Frees memory allocated by a Remote Desktop Services function.
+        /// </summary>
+        /// <param name="pMemory">Pointer to the memory to free.</param>
+        [DllImport(nameof(WtsApi32))]
+        public static extern void WTSFreeMemory(IntPtr pMemory);
+
+        /// <summary>
+        /// Retrieves a list of sessions on a Remote Desktop Session Host (RD Session Host) server.
+        /// </summary>
+        /// <param name="hServer">A handle to the RD Session Host server.
+        /// You can use the <see cref="WTSOpenServer"/> or <see cref="WTSOpenServerEx"/> functions to retrieve a handle to a specific
+        /// server, or <see cref="WTS_CURRENT_SERVER_HANDLE"/> to use the RD Session Host server that hosts your application.</param>
+        /// <param name="reserved">This parameter is reserved. It must be zero.</param>
+        /// <param name="version">The version of the enumeration request. This parameter must be 1.</param>
+        /// <param name="ppSessionInfo">A pointer to an array of <see cref="WTS_SESSION_INFO"/> structures that represent the retrieved
+        /// sessions. To free the returned buffer, call the <see cref="WTSFreeMemory"/> function.</param>
+        /// <param name="pSessionInfoCount">A pointer to the number of WTS_SESSION_INFO structures returned in the ppSessionInfo parameter.</param>
+        /// <returns>Returns zero if this function fails. If this function succeeds, a nonzero value is returned.</returns>
+        [DllImport(nameof(WtsApi32), SetLastError = true, CharSet = CharSet.Auto)]
+        public static extern bool WTSEnumerateSessions(
+            IntPtr hServer,
+            [MarshalAs(UnmanagedType.U4)] int reserved,
+            [MarshalAs(UnmanagedType.U4)] int version,
+            ref IntPtr ppSessionInfo,
+            [MarshalAs(UnmanagedType.U4)] ref int pSessionInfoCount);
```

Comment:
Please use `out int` (and drop the attribute) since this is an `_Out_` in native code, and does not need to be initialized by the managed code calling this method.